From fbdc9e8a69cc2b8380e4dff49fd2a96c39757a1b Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 24 Sep 2014 15:21:57 -0400
Subject: [infiniband] ocrdma: Convert to use simple_open()

Message-id: <13947e954e7876f85a13546e2cd292bc1587fb7c.1411571481.git.dledford@redhat.com>
Patchwork-id: 96616
O-Subject: [Patch RHEL7 22/48] RDMA/ocrdma: Convert to use simple_open()
Bugzilla: 1081067 1088577
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Duan Jiong <duanj.fnst@cn.fujitsu.com>

This removes an open-coded duplicate of simple_open().

Signed-off-by: Duan Jiong <duanj.fnst@cn.fujitsu.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 0cc65dd6918f529ae2c19be95b86dec19549b7ed)
Signed-off-by: Doug Ledford <dledford@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/ocrdma/ocrdma_stats.c b/drivers/infiniband/hw/ocrdma/ocrdma_stats.c
index 6c54106..41a9aec 100644
--- a/drivers/infiniband/hw/ocrdma/ocrdma_stats.c
+++ b/drivers/infiniband/hw/ocrdma/ocrdma_stats.c
@@ -510,16 +510,9 @@ exit:
  return status;
 }
 
-static int ocrdma_debugfs_open(struct inode *inode, struct file *file)
-{
- if (inode->i_private)
-  file->private_data = inode->i_private;
- return 0;
-}
-
 static const struct file_operations ocrdma_dbg_ops = {
  .owner = THIS_MODULE,
- .open = ocrdma_debugfs_open,
+ .open = simple_open,
  .read = ocrdma_dbgfs_ops_read,
 };
 
-- 
1.7.1